hazelcast/src/test/java/com/hazelcast/map/MapPutDestroyTest.java

```diff
@@ -0,0 +1,81 @@
+package com.hazelcast.map;
+
+
+import org.junit.Test;
+import com.hazelcast.core.IMap;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.CyclicBarrier;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.experimental.categories.Category;
+
+import java.util.concurrent.BrokenBarrierException;
+
+import com.hazelcast.test.HazelcastParallelClassRunner;
+
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category(QuickTest.class)
+public class MapPutDestroyTest extends HazelcastTestSupport {
+    @Test
+    public void testMapPutDestroy() {
```

Comment:
can you rename this test, currently it is hard to understand what it is testing
new name maybe something like `testConcurrentPutDestroy_doesNotCauseNPE `